### PR TITLE
Support for location based automation and multiple alarm zones

### PR DESCRIPTION
**Pull request by paolotremadio**

### CODE REVIEW

File: src/SecuritySystemAccessory.js [hunk 1]
```diff
@@ -59,10 +59,20 @@ class SecuritySystemAccessory {
...
       this.getAccessoryInformationService(),
       this.getBridgingStateService(),
-      this.getSecuritySystemService()
+      this.getSecuritySystemService(),
+      this.getArmSwitchService(),
     ];
   }
 
+  getArmSwitchService() {
+    this._armSwitchService = new Service.Switch(`${this.name} Arm`);
```
Comment (grover) [inline src/SecuritySystemAccessory.js]: Please pull this name from configuration for localization purposes.

File: src/SecuritySystemAccessory.js [hunk 2]
```diff
@@ -133,9 +158,14 @@ class SecuritySystemAccessory {
...
 
+  _isDisarmed() {
+    let currentState = this._state.targetState;
+    return currentState === Characteristic.SecuritySystemCurrentState.DISARMED;
+  }
+
   _updateCurrentState() {
     let currentState = this._state.targetState;
-    if (this._state.alarm && currentState !== Characteristic.SecuritySystemCurrentState.DISARMED) {
+    if (this._state.alarm && !this._isDisarmed()) {
```
Comment (grover) [inline src/SecuritySystemAccessory.js]: 👍🏻

File: src/SecuritySystemAccessory.js [hunk 3]
```diff
@@ -112,6 +122,21 @@ class SecuritySystemAccessory {
     this._persist(data, callback);
   }
 
+  _setArm(value, callback) {
+    let targetState;
+    if (value) {
+      targetState = Characteristic.SecuritySystemTargetState.AWAY_ARM;
```
Comment (grover) [inline src/SecuritySystemAccessory.js]: Might want to provide another switch for the home armed state?

File: src/SecuritySystemAccessory.js [hunk 4]
```diff
@@ -24,15 +24,27 @@ class SecuritySystemAccessory {
...
+      zonesAlarm: {},
     };
 
+    for (let zoneLabel of this.zones) {
+      defaultValue.zonesAlarm[zoneLabel] = false;
+    }
+
     storage.retrieve(defaultValue, (error, value) => {
+      // Remove zones from storage that are not in the list anymore
+      for (let zoneLabel in value.zonesAlarm) {
```
Comment (grover) [inline src/SecuritySystemAccessory.js]: Persist this change back to storage?

Comment (paolotremadio) [reply src/SecuritySystemAccessory.js]: Done :)

File: src/SecuritySystemAccessory.js (hunk 3 above)
Comment (paolotremadio) [reply src/SecuritySystemAccessory.js]: Done this too

**Comment by paolotremadio** [comment]
All done :)

**Pull request merged**